src/Particle/VirtualParticleSet.h

```diff
@@ -75,16 +75,19 @@ class VirtualParticleSet : public ParticleSet
 
   /** move virtual particles to new postions and update distance tables
      * @param jel reference particle that all the VP moves from
-     * @param ref_pos reference particle position
+     * @param p reference particle set
      * @param deltaV Position delta for virtual moves.
      * @param sphere set true if VP are on a sphere around the reference source particle
      * @param iat reference source particle
      */
-  void makeMoves(int jel,
-                 const PosType& ref_pos,
-                 const std::vector<PosType>& deltaV,
-                 bool sphere = false,
-                 int iat     = -1);
+  void makeMoves(int jel, const ParticleSet& p, const std::vector<PosType>& deltaV, bool sphere = false, int iat = -1);
```

Comment:
Only exposing ref_pos is preferred. Please add two argument ref_pos and ref_spinor instead of particleset. Also add function comment.

Reply:
done